### ISSUE TITLE
Implement Risk Metrics: Value At Risk 

### ISSUE THREAD

**Issue by AlexCatarino** [QuantConnect/Lean#6811]
#### Expected Behavior
Portfolio Statistics include Value at Risk (VaR) 

#### Actual Behavior
Not supported

#### Checklist
- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)
- [x] I have provided detailed steps to reproduce the issue

**Comment by wtindall1** [comment]
Hi @AlexCatarino - please could I pick up this issue?

**Comment by AlexCatarino** [comment]
Hi @wtindall1 , thank you!
I suggest adding your thoughts about this problem before working on it. 
VaR is a statistical metric, an "indicator" (see guidance on GitHub issues for Indicators), so we must create a unit test that compares the calculation with an external source.

**Comment by wtindall1** [comment]
When you say VaR should be an "indicator", do you mean that the scope of the issue is an implementation of `IIndicator`? Rather than a property of `PortfolioStatistics` (as I had assumed from the Expected Behaviour)

**Comment by AlexCatarino** [comment]
If VaR can be an `IIndicator` and a property of `PortfolioStatistics`, the better, right? :-)
For example, see Sharpe and Sortino ratios. 
If not, yes, it should be property of `PortfolioStatistics`. The point I was trying to make is that it should be validated like an indicator (external data file generated with third-party software, you need to tell us how it was created).

**Comment by wtindall1** [comment]
Makes sense - thank you. I will have a look into this and comment back here

**Comment by wtindall1** [comment]
@AlexCatarino - proposed script for generating the test data, based on:  https://blog.quantinsti.com/calculating-value-at-risk-in-excel-python/
[spy_var.csv](https://github.com/QuantConnect/Lean/files/14013792/spy_var.csv)

```
import pandas as pd
import numpy as np
from scipy.stats import norm

spy = pd.read_csv("https://github.com/QuantConnect/Lean/raw/master/Data/equity/usa/daily/spy.zip",
                       index_col=0, names=["open", "high", "low", "close", "volume"])

spy_close = spy[['close']]
spy['returns'] = spy_close.close.pct_change()

window_length = 252
spy['mean'] = spy['returns'].rolling(window_length).mean()
spy['stdev'] = spy['returns'].rolling(window_length).std()

spy['VaR_99'] = spy['mean'] - (norm.ppf(0.01) * spy['stdev'])
spy['VaR_95'] = spy['mean'] - (norm.ppf(0.05) * spy['stdev'])
spy['VaR_90'] = spy['mean'] - (norm.ppf(0.1) * spy['stdev'])

columns_to_export = ['close','VaR_99', 'VaR_95', 'VaR_90']

spy[columns_to_export].to_csv("spy_var.csv")
```
